inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/hudi/HudiCatalogClient.java

```diff
@@ -0,0 +1,244 @@
+        Table hiveTable = client.getTable(dbName, tableName);
+        Table newHiveTable = hiveTable.deepCopy();
+        List<FieldSchema> cols = newHiveTable.getSd().getCols();
+        for (HudiColumnInfo column : columns) {
+            FieldSchema fieldSchema = new FieldSchema();
+            fieldSchema.setName(column.getName());
+            fieldSchema.setType(column.getType());
+            fieldSchema.setComment(column.getDesc());
+            cols.add(fieldSchema);
+        }
+        newHiveTable.getSd().setCols(cols);
+        client.alter_table(dbName, tableName, newHiveTable);
+    }
+
+    /**
+     * Create hudi table and register to hive metastore
+     * @param tableName the hudi table name
+     * @param tableInfo the hudi table info
+     * @param useRealTimeInputFormat ignore uber input Format 
+     */
+    public void createTable(
+            String tableName,
+            HudiTableInfo tableInfo,
+            boolean useRealTimeInputFormat)
+            throws TException, IOException {
+        Table hiveTable = org.apache.hadoop.hive.ql.metadata.Table.getEmptyTable(dbName, tableName);
+        hiveTable.setOwner(UserGroupInformation.getCurrentUser().getUserName());
+        hiveTable.setCreateTime((int) (System.currentTimeMillis() / 1000));
+
+        Map<String, String> properties = new HashMap<>();
+        String location = this.warehouse + "/" + dbName + ".db" + "/" + tableName;
+        properties.put("path", location);
+
+        List<FieldSchema> cols = new ArrayList<>();
+        for (HudiColumnInfo column : tableInfo.getColumns()) {
+            FieldSchema fieldSchema = new FieldSchema();
+            fieldSchema.setName(column.getName());
+            fieldSchema.setType(column.getType());
+            fieldSchema.setComment(column.getDesc());
+            cols.add(fieldSchema);
+        }
+
+        // Build storage of hudi table
+        StorageDescriptor sd = new StorageDescriptor();
+        sd.setCols(cols);
+        hiveTable.setDbName(dbName);
+        hiveTable.setTableName(tableName);
+        // FIXME: splitSchemas need config by frontend
+
+        HoodieFileFormat baseFileFormat = HoodieFileFormat.PARQUET;
+        // ignore uber input Format
+        String inputFormatClassName =
+                HoodieInputFormatUtils.getInputFormatClassName(baseFileFormat, useRealTimeInputFormat);
+        String outputFormatClassName = HoodieInputFormatUtils.getOutputFormatClassName(baseFileFormat);
+        String serDeClassName = HoodieInputFormatUtils.getSerDeClassName(baseFileFormat);
+        sd.setInputFormat(inputFormatClassName);
+        sd.setOutputFormat(outputFormatClassName);
+        Map<String, String> serdeProperties = new HashMap<>();
+        serdeProperties.put("path", location);
+        serdeProperties.put(ConfigUtils.IS_QUERY_AS_RO_TABLE, String.valueOf(!useRealTimeInputFormat));
+        serdeProperties.put("serialization.format", "1");
```

Comment:
Why set the `format` to `1`? Does it have any value else?

Reply:
`serialization.format` is the field delimiter character in file between two column fields of the table when the file is serialized. This option does not take effect in the hudi table, so can cancel.